dongle_app/js/app.js

```diff
@@ -151,4 +151,23 @@ window.onload = function() {
       showQR(dongle);
     }
   );
+
+  function updateOnlineStatus(event) {
+    var ele = document.getElementById('not-on-line');
+    if(navigator.onLine){
+      ele.classList.remove('open');
+    }else{
+      ele.classList.add('open');
+    }
+  }
+
+  window.addEventListener('online',  updateOnlineStatus);
+  window.addEventListener('offline', updateOnlineStatus);
```

Comment:
I would change the approach here. You want to check at the beginning if there is connection, so a simple solution could be added here:
1) Check if onLine is true
2) If not, add a listener to execute 'register' function
3) If it's connected, execute register directly

`register` will be a wrapper of the lines you can find in [1]. I hope it helps!

[1] https://github.com/borjasalguero/foxcast/blob/master/dongle_app/js/app.js#L142